hazelcast/src/test/java/com/hazelcast/spi/utils/RestClientTest.java

```diff
@@ -280,35 +285,41 @@ static final class Tls13CipherCheckingServer implements Runnable {
             LOGGER.info("The server will be listening on port " + serverSocket.getLocalPort());
         }
 
+        public int getPort() {
+            return serverSocket.getLocalPort();
+        }
+
         public void run() {
-            try {
-                while (!(shutdownRequested || tls13CipherFound.get())) {
-                    try {
-                        Socket socket = serverSocket.accept();
-                        new Thread(() -> {
-                            LOGGER.info("Socket accepted " + socket);
-                            try {
-                                socket.setSoTimeout(5000);
-                                tls13CipherFound.compareAndSet(false, hasTls13Cipher(socket.getInputStream()));
-                            } catch (IOException e) {
-                                LOGGER.warning("Reading from the socket failed", e);
-                            } finally {
-                                close(socket);
-                            }
-                        }).start();
-                    } catch (SocketTimeoutException e) {
-                        // it's fine
-                    }
+            while (!(shutdownRequested || tls13CipherFound.get())) {
+                try {
+                    Socket socket = serverSocket.accept();
+                    new Thread(() -> {
+                        LOGGER.info("Socket accepted " + socket);
+                        try {
+                            socket.setSoTimeout(5000);
+                            tls13CipherFound.compareAndSet(false, hasTls13Cipher(socket.getInputStream()));
```

Comment:
IMHO volatile boolean + setting here = true (without compareAndSet) is enough, no need for AtomicBoolean

Reply:
The volatile field would be safe only in case you'd only set it `true` (i.e. either add an additional condition or `hasTls13Cipher()` throwing an exception instead of returning `false`).